conans/client/source.py

```diff
@@ -142,10 +157,11 @@ def _run_source(conanfile, conanfile_path, src_folder, hook_manager, reference,
                 output.info('Configuring sources in %s' % src_folder)
                 _run_scm(conanfile, src_folder, local_sources_path, output, cache=cache)
 
+                get_sources_from_exports()
+
                 if cache:
-                    _get_sources_from_exports(conanfile, src_folder, export_folder,
-                                              export_source_folder, cache)
-                    _clean_source_folder(src_folder)
+                    _clean_source_folder(src_folder)  # TODO: Why is it needed in cache?
```

Comment:
It can't be executed in the local workflow because it will remove the `conanfile.py` (if default source_folder, and conanfile in it). But I wonder why we want it in the cache, why? What's the problem of having those files there?

Reply:
I don't know.